framework/src/engine/bft/method.ts

```diff
@@ -98,22 +98,42 @@ export class BFTMethod {
 		};
 	}
 
-	// This function expects to be called after blockBFTInfos has been inserted for the executiong block.
-	// blockBFTInfos must be inserted first because for the block genesis block + 1, bftVotes.blockBFTInfos will be empty
-	public async currentHeaderImpliesMaximalPrevotes(stateStore: StateStore): Promise<boolean> {
+	public async impliesMaximalPrevotes(
```

Comment:
@janhack does this function looks ok?
I needed to update the function so that it can be called before/after the bftVotes has been updated

Reply:
I don't think this works. For instance, assume:
- header.height == 310
- header.maxHeightGenerated == 1
- MAX_LENGTH_BLOCK_BFT_INFOS == 309
Then, if you call `impliesMaximalPrevotes` before updating `bftVotes`, then you call check that bftVotes.blockBFTInfos[308].generatorAddress == header.generatorAddress` (this check could return True or False). However, if you do this check after updating `bftVotes`, then the previous block from the generator is not contained in `bftVotes.blockBFTInfos` any more.

Reply:
We could change the protocol however so that `impliesMaximalPrevotes` only uses the last `MAX_LENGTH_BLOCK_BFT_INFOS-1` instead of `MAX_LENGTH_BLOCK_BFT_INFOS` blocks.

Reply:
ah that's a good point, blockBFTInfos will be truncated at the edge.
since we add the property to block header, we can control when to call this function precisely, i will remove the logic to handle both and ensure it will be called before updating the bftVotes

Reply:
Okay, that sounds good. I resolve this comment then.